week-06-intro-ruby-on-rails/day-02/module-01/readme.md

```diff
@@ -1 +1,445 @@
+  * and `return`s the updated `contacts` hash.
+
+A sample contacts hash you could use to test would be `sample = {"Bob"=> 4152890045, "Emergency"=> 911, "Jenny"=> 8675309}`.
+
+1. Using `Array#map`, write a method called `get_responses` that takes an array of questions (strings) and returns an array of responses input from the console for each question. (Hint: you will need to use `gets.chomp` and `puts` ).
+
+**Challenge Set 3: Let's have a HTTParty!**
+
+4. Let's have an [HTTParty](https://github.com/jnunemaker/httparty)!
+
+5. Install the httparty gem `$ gem install httparty`.
+
+6. Now require it in a new ruby script file, and use it to call an album search on the word "White" to the spotify API.
+
+7. Can you require both `httparty` and `awesome_print` to have the output look nice? (remember just require awesome_print and then use `ap` instead of `p`)
+
+5. In the same file, can you write a loop that returns an array of the album names from your search?
+
+###Stretch Challenges
+
+1. Make your `is_prime?` method more efficient. Three hints on how to proceed are spaced out below. Before implementing each hint, develop a short argument to convince yourself that it works.  **Note: There are more challenges after the white space!**
+
+   <br><br><br><br><br>
+  * Hint: only check possible divisors up to half the original number
+   <br><br><br><br><br>
+  * Hint: only check possible divisors up to the square root of the original number
+    <br><br><br><br><br>
+  * Hint: if the number isn't 2, only check whether the number is divisible by odd numbers
+
+
+1. Find 3 gems you think are cool and try to use them in your own scripts. ([Ruby Toolbox](https://www.ruby-toolbox.com/)) has a lot of great gems listed, though you might have to hold off on incorporating Rails gems.
+
+1. Guessing Game
+
+   Create a program that asks the user to guess a number between 1 and 100.  Once the user guesses a number, the program should say higher or lower, or report that the number was correct.  The user should continue to make guesses until the correct number is found.  Also, once the user guesses correctly, the program should print the number of guesses needed to arrive at the correct answer. Below is sample output:
+
+   ```
+   Guess a number between 1 and 100
+   50
+   The number is lower than 50.  Guess again!
+   25
+   The number is lower than 25.  Guess again!
+   13
+   The number is higher than 13.  Guess again!
+   20
+   The number is lower than 20.  Guess again!
+   17
+   The number is higher than 17.  Guess again!
+   18
+   The number is higher than 18.  Guess again!
+   19
+   You got 19 in 7 tries!
+   ```
+
+1. Rewrite `factorial` without recursion. For an extra challenge, write a recursive *and* an iterative version of a method that takes in a number `num` and calculate the `num`th [Fibionacci number](https://en.wikipedia.org/wiki/Fibonacci_number).
```

Comment:
If you want them to do factorial AND fibonacci, maybe make this into two separate stretch challenges? Also, open link in new tab :)